parser/src/main/java/com/linkedin/avroutil1/parser/avsc/AvscParser.java

```diff
@@ -0,0 +1,162 @@
+            result.recordError(new JsonParseException("json parse error at " + e.getLocation(), e));
+            return result;
+        } catch (Exception e) {
+            result.recordError(new JsonParseException("unknown json parse error", e));
+            return result;
+        }
+
+        try {
+            parseSchemaDeclOrRef(root, context, true);
+            result.recordParseComplete(context);
+        } catch (Exception parseIssue) {
+            result.recordError(parseIssue);
+        }
+
+        return result;
+    }
+
+    /**
+     * parses a schema (or a reference to one by FQCN) out of a given json node
+     * @param node json node (which is a schema declaration or reference)
+     * @param context context for this parse operation
+     * @param topLevel true if this schema is the "top-level" schema for this context
+     *                 (for example outer-most schema in an avsc file)
+     * @throws ParseException
+     */
+    private void parseSchemaDeclOrRef (
+            JsonObjectExt node,
+            AvscParseContext context,
+            boolean topLevel
+    ) throws ParseException {
+        JsonValue.ValueType nodeType = node.getValueType();
+        AvroSchema definedSchema;
+        switch (nodeType) {
+            case STRING: //primitive or ref
+                throw new UnsupportedOperationException("TBD");
+            case OBJECT: //record/enum/fixed/error
+                Located<String> typeStr = getRequiredString(node, "type", () -> "it is a schema declaration");
+                AvroType avroType = AvroType.fromJson(typeStr.getValue());
+                if (avroType == null) {
+                    throw new AvroSyntaxException("unknown avro type \"" + typeStr.getValue() + "\" at " + typeStr.getLocation() + ". expecting \"record\", \"enum\" or \"fixed\"");
+                }
+                Located<String> nameStr = null;
+                Located<String> namespaceStr = null;
+                Located<String> docStr = null;
+                String namespace = context.getCurrentNamespace(); // != null
+                boolean namespaceChanged = false;
+                String doc = null;
+                if (avroType.isNamed()) {
+                    nameStr = getRequiredString(node, "name", () -> avroType + " is a named type");
+
+                    namespaceStr = getOptionalString(node, "namespace");
+                    //check if context namespace changed
+                    if (namespaceStr != null) {
+                        String newNamespace = namespaceStr.getValue();
+                        if (!namespace.equals(newNamespace)) {
+                            context.pushNamespace(newNamespace);
+                            namespaceChanged = true;
+                            namespace = newNamespace;
+                        }
+                    }
+
+                    //technically the avro spec does not allow "doc" on type fixed, but screw that
```

Comment:
Aside from the comments, can you perhaps add a README or some doc on how this parser differs from the avro parsed? As you build your parser, you can add more information. This will collate the screw avro comments  that is dispersed across the codebase. :)

Reply:
beefed up the javadocs on the class. a full RADME for the module will happen when more of the module exists

Reply:
Thank you!